pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -42,26 +45,63 @@ func (a Adapter) Start() (err error) {
 
 	objectMeta := kclient.CreateObjectMeta(componentName, a.Client.Namespace, labels, nil)
```

Comment:
We're probably unlikely to hit this issue with component names, **but** to be safe, we should probably trim and sanitize `componentName` with either `utils.DNS1123Name()` or `utils.NamespaceOpenShiftObject` before passing it into CreateObjectMeta.

Reply:
updated